internal/infrastructure/middleware/rbac.go

```diff
@@ -0,0 +1,33 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	echo_middleware "github.com/labstack/echo/v4/middleware"
+)
+
+// TODO HMS-3522 Add RBACMap and implement its methods:
+//
+// Add(route string, method string, permission string) RBACMap
+// Get(route string, method string) string
+type RBACMap map[string]map[string]string
```

Comment:
Please document what the RBACMap contains. Is it `route -> method -> permission` or `method -> route -> permission`?